packages/core/src/crypto/jose/jwk/P_521Jwk.ts

```diff
@@ -0,0 +1,102 @@
+import type { JwkJson } from './Jwk'
+
+import { TypedArrayEncoder, Buffer } from '../../../utils'
+import { KeyType } from '../../KeyType'
+import { JwaCurve, JwaKeyType } from '../jwa'
+import { JwaSignatureAlgorithm } from '../jwa/alg'
+
+import { Jwk } from './Jwk'
+import { compress, expand } from './ecCompression'
+import { hasKty, hasCrv, hasX, hasY, hasValidUse } from './validate'
+
+export class P_521Jwk extends Jwk {
+  public readonly x: string
+  public readonly y: string
+
+  public constructor({ x, y }: { x: string; y: string }) {
+    super()
+
+    this.x = x
+    this.y = y
+  }
+
+  public get kty() {
+    return JwaKeyType.EC as const
+  }
+
+  public get crv() {
+    return JwaCurve.P_521 as const
+  }
+
+  public get keyType() {
+    return KeyType.P521
+  }
+
+  // NOTE: this is the compressed variant. We should add support for the uncompressed variant.
+  public get publicKey() {
+    const publicKeyBuffer = Buffer.concat([TypedArrayEncoder.fromBase64(this.x), TypedArrayEncoder.fromBase64(this.y)])
+    const compressedPublicKey = compress(publicKeyBuffer)
+
+    return Buffer.from(compressedPublicKey)
+  }
+
+  public get supportedEncryptionAlgorithms() {
+    return []
+  }
+
+  public get supportedSignatureAlgorithms() {
+    return [JwaSignatureAlgorithm.ES512]
```

Comment:
EsXXX also has a matching hashing algorithm. Should we also add that as a `supportedHashingAlgorithm?

ES256 -> Sha2-256
ES384 -> Sha2-384
ES521 -> Sha2-512

Reply:
Can you point me to the spec where it defines ES as a hashing algorithm by itself?

Reply:
Ah no you are correct, ignore my comment :).